multiviewer-for-f1: add update script

bglebrun opened the pull request:
## Description of changes

Add update script to multiviewer-for-f1 derivation

Possibly closes #336005

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

bglebrun committed:
Added update script for multiviewer-for-f1

bglebrun commented:
After pull, will:

bglebrun commented:
Unless we want the latest in master before this GP weekend, wonder if we could let the script do it's update and close #336005 for now

babeuh commented:
Okay so my bad, I misread your lest message  on the previous PR. I actually meant that you should have one commit for the version bump and one commit for the script inside one single PR. I should've been clearer with that too

It doesn't matter though, this works as well!

Anyways, this looks good to me, I don't have access to my NixOS desktop right now to test it but I don't see anything wrong

bglebrun commented:
I also figured unless the package is currently broken, the script can do the bump the next time the repo-wide update is triggered.

bglebrun commented:
> Also I think you might have to change the PR name to something like
> ```multiviewer-for-f1: add update script```
Fixed

FliegendeWurst commented on pkgs/applications/video/multiviewer-for-f1/default.nix:
```diff
@@ -90,6 +91,27 @@ stdenvNoCC.mkDerivation rec {
+    link=$(echo $latest | jq -r '.downloads[] | select(.platform=="linux_deb").url')
+    id=$(echo $latest | jq -r '.downloads[] | select(.platform=="linux_deb").id')
+    version=$(echo $latest | jq -r '.version')
+
```
I think it would be good to skip the download if the version is already packaged. No need to wastefully download ~150MB.

bglebrun committed:
Fix update script to perform check

Checks if the version pulled from our metadata API is the same as the current version, preventing a pull of the package if an update is not necessary.

FliegendeWurst commented:
Please fix the merge conflict.

bglebrun committed:
Resolved upstream merge issue

bglebrun committed:
nixfmt fix

bglebrun committed:
nixfmt

bglebrun committed:
Merge branch 'multiviewer-for-f1-update-script' of github.com:bglebrun/nixpkgs into multiviewer-for-f1-update-script

bglebrun reviewed (commented):
Script should now check for current version to latest and omit the package pull and update if no update is necessary

linuxmobile commented:
Why this is stucked and outdated?

bglebrun commented:
I've been waiting, as far as I know everything is taken care of. Would probably be faster to open a pull request for an update.

babeuh reviewed (approved):
Still looks good to me

linuxmobile commented:
Merge this before my PR will break my PR I think. https://github.com/NixOS/nixpkgs/pull/391691
